test/basic/elasticache/elasticache_test.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/aws/aws-sdk-go-v2/service/elasticache/types"
+
+	//"github.com/aws/aws-sdk-go/aws/awserr"
+	//"github.com/aws/aws-sdk-go/aws/session"
+	legacy_aws_sdk_ec2 "github.com/aws/aws-sdk-go/service/ec2"
+
+	terratest_aws "github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+
+	cac "github.com/aptible/cloud-api-clients/clients/go"
+	"github.com/aptible/terraform-provider-aptible-iaas/internal/client"
+)
+
+func cleanupAndAssert(t *testing.T, terraformOptions *terraform.Options) {
+	terraform.Destroy(t, terraformOptions)
+
+	// test / assert all failures here
+}
+
+func TestElasticache(t *testing.T) {
+	vpc_name := "test-elasticache-vpc"
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: ".",
+
+		Vars: map[string]interface{}{
+			"organization_id":    os.Getenv("ORGANIZATION_ID"),
+			"environment_id":     os.Getenv("ENVIRONMENT_ID"),
+			"aptible_host":       os.Getenv("APTIBLE_HOST"),
+			"node_name":          "test-elasticache-node",
+			"vpc_name":           vpc_name,
+			"description":        "TestingModule",
+			"snapshot_window":    "01:00-04:00",
+			"maintenance_window": "sun:05:00-sun:09:00",
+		},
+	})
+
+	// Make sure to delete when the function returns.
+	defer cleanupAndAssert(t, terraformOptions)
+
+	// Run the Terraform Workspace.
+	terraform.InitAndApply(t, terraformOptions)
+
+	// Initialize the Cloud-API Client
+	c := client.NewClient(
+		true,
+		os.Getenv("APTIBLE_HOST"),
+		os.Getenv("APTIBLE_TOKEN"),
+	)
+	ctx := context.Background()
+
+	// Get VPC Asset ID from terraform output and use it to get the Asset from Cloud-API
+	vpcId := terraform.Output(t, terraformOptions, "vpc_id")
+
+	// check cloud api's understanding of asset
+	vpcAsset, vpcAptibleErr := c.DescribeAsset(
+		ctx,
+		os.Getenv("ORGANIZATION_ID"),
+		os.Getenv("ENVIRONMENT_ID"),
+		vpcId,
+	)
+	assert.Nil(t, vpcAptibleErr)
+	assert.Equal(t, vpcAsset.Id, vpcId)
+	assert.Equal(t, vpcAsset.Status, cac.ASSETSTATUS_DEPLOYED)
+
+	// check aws asset state
+	vpcAws, vpcAwsErr := terratest_aws.GetVpcsE(t, []*legacy_aws_sdk_ec2.Filter{
+		{
+			Name:   aws.String("tag:Name"),
+			Values: []*string{aws.String(vpc_name)},
+		},
+	}, "us-east-1")
+	assert.Nil(t, vpcAwsErr)
+	assert.GreaterOrEqual(t, len(vpcAws), 1)
+	assert.Equal(t, len(vpcAws[0].Subnets), 6)
+
+	redisId := terraform.Output(t, terraformOptions, "redis_id")
+
+	// check cloud api's understanding of asset
+	redisAsset, redisErr := c.DescribeAsset(
+		ctx,
+		os.Getenv("ORGANIZATION_ID"),
+		os.Getenv("ENVIRONMENT_ID"),
+		redisId,
+	)
+	assert.Nil(t, redisErr)
+	assert.Equal(t, redisAsset.Id, redisId)
+	assert.Equal(t, redisAsset.Status, cac.ASSETSTATUS_DEPLOYED)
+	assert.NotNil(t, redisAsset.Outputs)
+
+	redis_cluster_id_base := redisAsset.GetOutputs()["elasticache_cluster_id"].Data.(string)
+
+	cluster_details, err := GetCluster(redis_cluster_id_base + "-001")
```

Comment:
i think our cluster id output is probably wrong from the terraform and we're sending up a different id

```
output "elasticache_cluster_id" {
  value = module.elasticache_redis.aws_elasticache_cluster.id
}
```

maybe we want a different id here? `aws_elasticache_replication_group` should be output on the asset bundle instead maybe? or something else? i think is what we might actually want: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_replication_group#member_clusters

and then query on these

Reply:
follow up ticket made, resolving